OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/config/impl/ConfigModelStoreListener.kt

```diff
@@ -72,7 +72,7 @@ internal class ConfigModelStoreListener(
                     // copy current model into new model, then override with what comes down.
                     val config = ConfigModel()
                     config.initializeFromModel(null, _configModelStore.model)
-
+                    config.fetchParamsNotifier = _configModelStore.model.fetchParamsNotifier
```

Comment:
This is actually to prevent just that. It moves the notifier from the older config model into a new one in the event of parameter fetched so the notifier that was added before can get fired even the config model has been replaced.

Reply:
I see, you are right, I mixed up what was happening here.

After looking at this again this extra event seems unnecessary, as we already have a `onModelReplaced` event that can be listen to instead.

Reply:
I think the `IdentityVerificationService` you created is the correct pattern, however I believe it could have just used `onModelReplaced`, instead of needing to introduce `fetchParamsNotifier`. Switching to that means less code and follow other patterns.
* However if your `fetchParamsNotifier` handles something I am not seeing let me know, or you feel it is a better pattern here.